Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -244,8 +244,10 @@ public NodejsProjectNode(NodejsProjectPackage package)
         }
 
         protected override void FinishProjectCreation(string sourceFolder, string destFolder) {
+            bool foundTypeScriptFile = false;
             foreach (MSBuild.ProjectItem item in this.BuildProject.Items) {
                 if (String.Equals(Path.GetExtension(item.EvaluatedInclude), NodejsConstants.TypeScriptExtension, StringComparison.OrdinalIgnoreCase)) {
+                    foundTypeScriptFile = true;
```

Comment:
Also let's exclude node_modules and bower_components from his search since that shouldn't have any bearing on whether the project is a TS project. 

Additionally, we may want to consider just checking if EnableTypeScript flag is set since we explicitly have separate TS and JS project types.